common/src/main/java/com/wynntils/screens/maps/PoiManagementScreen.java

```diff
@@ -67,79 +111,756 @@ public void onClose() {
+
+            updateState = true;
+        }
+
+        selectAllButton.active = selectedPois.size() < pois.size();
+
+        if (updateState) {
+            selectionMode = add;
+
+            deselectAllButton.active = add;
+            deleteSelectedButton.active = add;
+
+            upButton.visible = add;
+            downButton.visible = add;
+
+            // Export tooltip should display how many of the pois will be exported
+            Component tooltip = add
+                    ? Component.translatable(
+                            "screens.wynntils.poiManagementGui.exportSelected.tooltip", selectedPois.size())
+                    : Component.translatable("screens.wynntils.poiManagementGui.exportAll.tooltip");
+
+            exportButton.setTooltip(Tooltip.create(tooltip));
+        }
+
+        populatePois();
+    }
+
+    public void toggleSortType(PoiSortType sortType, PoiSortButton selectedButton) {
+        if (activeSortButton != null && activeSortButton != selectedButton) {
+            activeSortButton.setSelected(false);
+        }
+
+        activeSortButton = selectedButton;
+
+        boolean selected = toggleSortOrder(sortType);
+
+        selectedButton.setSelected(selected);
+
+        populatePois();
+    }
+
+    public void deletePoi(CustomPoi poiToDelete) {
+        HiddenConfig<List<CustomPoi>> customPois = Managers.Feature.getFeatureInstance(MainMapFeature.class).customPois;
+        int deletedPoiIndex = customPois.get().indexOf(poiToDelete);
+
+        customPois.get().remove(poiToDelete);
+        customPois.touched();
+
+        deletedPois.add(poiToDelete);
+        deletedIndexes.add(deletedPoiIndex);
+
+        undoDeleteButton.active = true;
+
+        // Handles deleting when the final poi is visible
+        if (scrollOffset == Math.max(0, pois.size() - maxPoisToDisplay)) {
+            scroll(-1);
+        }
+
+        updateAllUsedIcons();
+
+        populatePois();
+    }
+
+    public void updatePoiPosition(CustomPoi poiToMove, int direction) {
+        int poiToMoveIndex = pois.indexOf(poiToMove);
+
+        // If poi is at the top/bottom of list or if it was selected but then filtered out, don't move it
+        if (poiToMoveIndex == -1 || poiToMoveIndex + direction < 0 || poiToMoveIndex + direction > pois.size() - 1) {
+            return;
+        }
+
+        CustomPoi poiToSwap = pois.get(pois.indexOf(poiToMove) + direction);
+
+        HiddenConfig<List<CustomPoi>> customPois = Managers.Feature.getFeatureInstance(MainMapFeature.class).customPois;
+
+        Collections.swap(
+                customPois.get(),
+                customPois.get().indexOf(poiToMove),
+                customPois.get().indexOf(poiToSwap));
+
+        customPois.touched();
+
+        populatePois();
+    }
+
+    @Override
+    public boolean doMouseClicked(double mouseX, double mouseY, int button) {
+        // FIXME: Search bar cursor dissapears after interacting with it but still remains focused
```

Comment:
I can look into this if needed

Reply:
If you wouldn't mind I would appreciate that, probably something simple I just can't see